reflex/experimental/vars/base.py

```diff
@@ -452,3 +465,241 @@ def create(
             _var_value=value,
             _var_data=_var_data,
         )
+
+
+@dataclasses.dataclass(
+    eq=False,
+    frozen=True,
+    **{"slots": True} if sys.version_info >= (3, 10) else {},
+)
+class LiteralBooleanVar(LiteralVar):
+    """Base class for immutable literal boolean vars."""
+
+    _var_value: bool = dataclasses.field(default=False)
+
+    @classmethod
+    def create(
+        cls,
+        value: bool,
+        _var_data: VarData | None = None,
+    ) -> LiteralBooleanVar:
+        """Create a var from a boolean value.
+
+        Args:
+            value: The value to create the var from.
+            _var_data: Additional hooks and imports associated with the Var.
+
+        Returns:
+            The var.
+        """
+        return LiteralBooleanVar(
+            _var_name="true" if value else "false",
+            _var_data=ImmutableVarData.merge(_var_data),
+            _var_type=bool,
+            _var_value=value,
+        )
+
+
+@dataclasses.dataclass(
+    eq=False,
+    frozen=True,
+    **{"slots": True} if sys.version_info >= (3, 10) else {},
+)
+class LiteralNumberVar(LiteralVar):
+    """Base class for immutable literal number vars."""
+
+    _var_value: float | int = dataclasses.field(default=0)
+
+    @classmethod
+    def create(
+        cls,
+        value: float | int,
+        _var_data: VarData | None = None,
+    ) -> LiteralNumberVar:
+        """Create a var from a number value.
+
+        Args:
+            value: The value to create the var from.
+            _var_data: Additional hooks and imports associated with the Var.
+
+        Returns:
+            The var.
+        """
+        return LiteralNumberVar(
+            _var_name=str(value),
+            _var_data=ImmutableVarData.merge(_var_data),
+            _var_type=float | int,
+            _var_value=value,
+        )
+
+
+@dataclasses.dataclass(
+    eq=False,
+    frozen=True,
+    **{"slots": True} if sys.version_info >= (3, 10) else {},
+)
+class LiteralObjectVar(LiteralVar):
+    """Base class for immutable literal object vars."""
+
+    _var_value: Tuple[Tuple[str, Var], ...] = dataclasses.field(default_factory=tuple)
```

Comment:
should both key and value should be `Var` type to support indirection of the key?

if we do it this way, then the js literal would put square brackets around the key.